src/Simplex/Chat/Controller.hs

```diff
@@ -624,6 +626,29 @@ instance ToJSON ChatResponse where
   toJSON = J.genericToJSON . sumTypeJSON $ dropPrefix "CR"
   toEncoding = J.genericToEncoding . sumTypeJSON $ dropPrefix "CR"
 
+-- reduce to 4 result types, add connection type to result?
+-- results may diverge per type, for example:
+-- CRUInvitationLinkKnown shouldn't be possible, because conn_req_inv is removed on connection status changing to ready;
+-- own link and own address / group link are checks for different things
+data ConnReqURICheckResult
+  = CRUInvitationLinkOkToConnect
+  | CRUInvitationLinkOwn
+  | CRUInvitationLinkConnecting {contact_ :: Maybe Contact}
+  | CRUInvitationLinkKnown {contact :: Contact}
+  | CRUContactAddressOkToConnect
+  | CRUContactAddressOwn
+  | CRUContactAddressConnecting {contact_ :: Maybe Contact}
+  | CRUContactAddressKnown {contact :: Contact}
+  | CRUGroupLinkOkToConnect
+  | CRUGroupLinkOwn
+  | CRUGroupLinkConnecting {groupInfo_ :: Maybe GroupInfo}
+  | CRUGroupLinkKnown {groupInfo :: GroupInfo}
+  deriving (Show, Generic)
```

Comment:
```suggestion
data ConnAction
  = CAOk
  | CAOwnLink
  | CAConnecting
  | CAKnown
  
data LinkType
  = LTInvitation {contact_:: Maybe Contact}
  | LTContact {contact_:: Maybe Contact}
  | LTGroup {groupInfo_ :: Maybe GroupInfo}
```

I understand the space is redundant now... Need to think.

Reply:
reworked into sum type